### PR TITLE
9119 - PDF Alternative

### PR DESCRIPTION
**Pull request by codyseibert**
fixing an issue with the maintenance mode storing the flag in localstorage which means users that were not using the application when the maintenance mode was disengaged would always have the flag as true.

### CODE REVIEW

File: docs/environments/setup.md [hunk 1]
```diff
@@ -131,41 +131,61 @@ EF-CMS currently has both the concept of a deployment at a domain as well as a n
     aws dynamodb put-item --region us-east-1 --table-name "efcms-deploy-${ENVIRONMENT}" --item '{"pk":{"S":"current-color"},"sk":{"S":"current-color"},"current":{"S":"blue"}}'
     ```
 
-13. Setup the environment's order search flag:
+13. Setup the environment's internal order search flag:
```
Comment (rachelschneiderman) [inline docs/environments/setup.md:134]: Nit: can these steps now be replaced by running `scripts/setup-all-env-configuration.sh`?

File: shared/src/persistence/dynamo/deployTable/getFeatureFlagValue.js [hunk 2]
```diff
@@ -9,14 +9,11 @@ const client = require('../../dynamodbClientService');
  * @returns {boolean} the boolean of the feature flag value
```
Comment (rachelschneiderman) [inline shared/src/persistence/dynamo/deployTable/getFeatureFlagValue.js]: Nit: docs need a small update

File: web-api/src/genericHandler.test.js [hunk 3]
```diff
@@ -266,11 +266,21 @@ describe('genericHandler', () => {
...
 
-      await expect(
-        checkMaintenanceMode({ applicationContext }),
-      ).resolves.not.toThrow();
+      await expect(checkMaintenanceMode({ applicationContext })).resolves.toBe(
+        false,
+      );
+    });
+
+    it('should throw an error if maintenance mode is true', async () => {
```
Comment (rachelschneiderman) [inline web-api/src/genericHandler.test.js]: Nit: test name seems copy/pasta'd

File: web-client/src/presenter/actions/clearMaintenanceModeAction.js [hunk 4]
```diff
@@ -1,14 +1,5 @@
 import { state } from 'cerebral';
 
-export const clearMaintenanceModeAction = async ({
-  applicationContext,
-  store,
-}) => {
+export const clearMaintenanceModeAction = async ({ store }) => {
```
Comment (rachelschneiderman) [inline web-client/src/presenter/actions/clearMaintenanceModeAction.js]: Nit: I don't think this has to be async anymore

File: web-client/src/presenter/actions/isTokenOnState.js [hunk 5]
```diff
@@ -0,0 +1,18 @@
...
+/**
+ * invokes the path in the sequences depending on if the user is logged in or not
+ *
+ * @param {object} providers the providers object
+ * @param {object} providers.path the cerebral path which is contains the next paths that can be invoked
+ * @param {object} providers.get the cerebral get method
+ * @returns {object} the list of section work items
+ */
+export const isTokenOnState = ({ get, path }) => {
+  const token = get(state.token);
```
Comment (rachelschneiderman) [inline web-client/src/presenter/actions/isTokenOnState.js]: Out of curiosity, I see we have `web-client/src/presenter/actions/isLoggedInAction.js`. Which checks if the user is defined in appContext. Is there an advantage to checking for the token instead? Are both needed?

If we decide to keep this action, could you please add a test file?

File: web-client/src/presenter/actions/setMaintenanceModeAction.js [hunk 6]
```diff
@@ -7,15 +7,6 @@ import { state } from 'cerebral';
  * @param {object} props the cerebral props object
  * @param {object} store the cerebral store
  */
-export const setMaintenanceModeAction = async ({
-  applicationContext,
-  props,
-  store,
-}) => {
+export const setMaintenanceModeAction = async ({ props, store }) => {
```
Comment (rachelschneiderman) [inline web-client/src/presenter/actions/setMaintenanceModeAction.js]: Nit: remove async keyword

File: web-client/src/presenter/actions/setUserAction.js [hunk 7]
```diff
@@ -13,6 +13,8 @@ import { state } from 'cerebral';
 export const setUserAction = async ({ applicationContext, props, store }) => {
   store.set(state.user, props.user);
   applicationContext.setCurrentUser(props.user);
+  const userPermissions = applicationContext.getCurrentUserPermissions();
```
Comment (rachelschneiderman) [inline web-client/src/presenter/actions/setUserAction.js:16]: Nit: Can you add a test around this please?

File: web-client/src/router.js [hunk 8]
```diff
@@ -49,14 +51,19 @@ const goto404 = app => {
...
+  { app, permissionToCheck, redirect = accessRedirects, skipMaintenanceCheck },
   cb,
 ) => {
   return function () {
-    if (!app.getState('user')) {
+    if (!app.getState('token')) {
       return redirect.gotoLoginPage(app);
     } else if (app.getState('maintenanceMode')) {
-      return redirect.gotoMaintenancePage(app);
+      if (!skipMaintenanceCheck) {
```
Comment (rachelschneiderman) [inline web-client/src/router.js:61]: Can you elaborate on why `skipMaintenanceCheck` was added here please?

Comment (codyseibert) [reply web-client/src/router.js:61]: So there is a bug with riot-router (or with cerebral?) where when you navigate to /maintenance while maintenanceMode is true on our state, our ifHasAccess check will also try to navigate to /maintenance which will cause the UI to not show anything except a spinner.  There might be a better fix for this, but I can't figure out how to force it to load the route.

**Review by mrinsin — APPROVED** [review]
Addressed our questions on zoom, `isLoggedInAction` to be refactored as DOD, approved and good to merge after tests are added and docs updated!

**Pull request merged**